Makefile

```diff
@@ -1,125 +1,152 @@
-$(VENV)/bin/activate:
-	@$(RUN_TASK) setup_venv "$(VENV)"
+lint: venv anchore-ci ## Lint code (currently using flake8)
+	@$(ACTIVATE_VENV) && $(CI_CMD) lint
+
+# Local CI script
+build: Dockerfile anchore-ci venv ## Build dev Anchore CLI Docker image
+	@$(CI_CMD) scripts/ci/build "$(COMMIT_SHA)" "$(GIT_REPO)" "$(TEST_IMAGE_NAME)"
+
+test: ## Run all tests: unit, functional, and e2e
+	@$(MAKE) test-unit
+	@$(MAKE) test-functional
+	@$(MAKE) test-e2e
+
+test-unit: anchore-ci venv ## Run unit tests (tox)
+	@$(ACTIVATE_VENV) && $(CI_CMD) test-unit
 
-.PHONY: install
-install: venv setup.py requirements.txt ## install project to virtual environment
-	@$(RUN_TASK) install "$(GIT_REPO)" "$(PYTHON)" "$(VENV)"
+test-functional: anchore-ci venv ## Run functional tests (tox)
+	@$(ACTIVATE_VENV) && $(CI_CMD) test-functional
 
-.PHONY: install-dev
-install-dev: venv setup.py requirements.txt ## install project to virtual environment in editable mode
-	@$(RUN_TASK) install_dev "$(GIT_REPO)" "$(PYTHON)" "$(VENV)"
+# Local CI scripts (set-e2e-tests and e2e-tests)
+test-e2e: anchore-ci venv ## Set up and run end to end tests
+test-e2e: CLUSTER_CONFIG := tests/e2e/kind-config.yaml
+test-e2e: KUBERNETES_VERSION := 1.15.7
+test-e2e: tests/e2e/kind-config.yaml
+	$(CI_CMD) install-cluster-deps "$(VENV)"
+	$(ACTIVATE_VENV) && $(CI_CMD) cluster-up "$(CLUSTER_NAME)" "$(CLUSTER_CONFIG)" "$(KUBERNETES_VERSION)"
+	$(ACTIVATE_VENV) && $(CI_CMD) setup-e2e-tests "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_TAG)" "$(TEST_IMAGE_NAME)"
+	$(ACTIVATE_VENV) && $(CI_CMD) e2e-tests
+	$(ACTIVATE_VENV) && $(CI_CMD) cluster-down "$(CLUSTER_NAME)"
 
-.PHONY: lint
-lint: venv ## lint code using pylint
-	@$(RUN_TASK) lint "$(PYTHON)" "$(VENV)"
+push-dev: anchore-ci ## Push dev Anchore CLI Docker image to Docker Hub
+	@$(CI_CMD) push-dev-image "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_BRANCH)" "$(TEST_IMAGE_NAME)"
 
-.PHONY: test
-test: test-unit test-functional test-e2e ## run all test make recipes -- test-unit, test-functional, test-e2e
+push-rc: anchore-ci ## Push RC Anchore CLI Docker image to Docker Hub (not available outside of CI)
+	@$(CI_CMD) push-rc-image "$(COMMIT_SHA)" "$(DEV_IMAGE_REPO)" "$(GIT_TAG)"
 
-.PHONY: test-unit
-test-unit: venv
-	@$(RUN_TASK) unit_tests "$(PYTHON)" "$(VENV)"
+push-prod: anchore-ci ## Push release Anchore CLI Docker image to Docker Hub (not available outside of CI)
+	@$(CI_CMD) push-prod-image-release "$(DEV_IMAGE_REPO)" "$(GIT_BRANCH)" "$(GIT_TAG)" "$(PROD_IMAGE_REPO)"
 
-.PHONY: test-functional
-test-functional: venv
-	@$(RUN_TASK) functional_tests "$(PYTHON)" "$(VENV)"
+push-rebuild: anchore-ci ## Rebuild and push prod Anchore CLI docker image to Docker Hub (not available outside of CI)
+	@$(CI_CMD) push-prod-image-rebuild "$(DEV_IMAGE_REPO)" "$(GIT_BRANCH)" "$(GIT_TAG)" "$(PROD_IMAGE_REPO)"
 
-.PHONY: test-e2e
-test-e2e: setup-test-e2e
-	@$(MAKE) run-test-e2e
-	@$(MAKE) cluster-down
+dist-deb: ## Package Anchore CLI for Debian-based distros
```

Comment:
All these dist targets can be deleted. We only push to Dockerhub & Pypi

Reply:
Cool!